core/src/appleMain/kotlin/ScannerBuilder.kt

```diff
@@ -34,8 +35,8 @@ public actual class ScannerBuilder {
         allowDuplicateKeys?.also {
             options[CBCentralManagerScanOptionAllowDuplicatesKey] = it
         }
-        solicitedServiceUuids?.also {
-            options[CBCentralManagerScanOptionSolicitedServiceUUIDsKey] = it.toTypedArray()
+        solicitedServiceUuids?.also { uuids ->
+            options[CBCentralManagerScanOptionSolicitedServiceUUIDsKey] = uuids.map { it.toCBUUID() } as NSArray
```

Comment:
`map` on `List<T>` will return another `List` which would probably (should?) fail to typecast here. Probably it should be:
```kotlin
    uuids.map { it.toCBUUID() }.toTypedArray()
```

My understanding is the `toTypedArray` will convert `List` to the native array type, certainly that is how it works when bridging arrays on Js for example, and one assumes it is the same for Obj-C.

Reply:
This causes a crash:

```
*** Terminating app due to uncaught exception 'Invalid Object', reason: 'Object kotlin.Array@94b0a00 is invalid'
```

Reply:
Wild.

Reply:
🤷‍♂️